test/e2e_env/kubernetes/meshpassthrough/meshpassthrough.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshPassthrough() {
+	const meshName = "mesh-passthrough"
+	const mesNamespace = "mesh-passthrough-mes"
+	const namespace = "mesh-passthrough"
+	var anotherEsIP string
+	var notAccessibleEsIP string
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MeshKubernetes(meshName)).
+			Install(Namespace(mesNamespace)).
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(
+				democlient.WithNamespace(namespace),
+				democlient.WithMesh(meshName),
+			)).
+			Install(testserver.Install(
+				testserver.WithNamespace(mesNamespace),
+				testserver.WithName("external-service"),
+			)).
+			Install(testserver.Install(
+				testserver.WithNamespace(mesNamespace),
+				testserver.WithName("another-external-service"),
+			)).
+			Install(testserver.Install(
+				testserver.WithNamespace(mesNamespace),
+				testserver.WithName("not-accessible-external-service"),
+			)).
+			Setup(kubernetes.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+		anotherEsIP, err = PodIPOfApp(kubernetes.Cluster, "another-external-service", mesNamespace)
+		Expect(err).ToNot(HaveOccurred())
+		notAccessibleEsIP, err = PodIPOfApp(kubernetes.Cluster, "not-accessible-external-service", mesNamespace)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEachFailure(func() {
+		DebugKube(kubernetes.Cluster, meshName, namespace)
+	})
+
+	E2EAfterEach(func() {
+		Expect(DeleteMeshResources(kubernetes.Cluster, meshName, v1alpha1.MeshPassthroughResourceTypeDescriptor)).To(Succeed())
+	})
+
+	E2EAfterAll(func() {
+		Expect(kubernetes.Cluster.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(kubernetes.Cluster.DeleteMesh(meshName)).To(Succeed())
+	})
+
+	It("should control passthrough cluster", func() {
+		// given
+		Eventually(func(g Gomega) {
+			_, err := client.CollectEchoResponse(
+				kubernetes.Cluster, "demo-client", "external-service.mesh-passthrough-mes.svc.cluster.local:80",
+				client.FromKubernetesPod(namespace, "demo-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+		}, "30s", "1s").Should(Succeed())
+
+		meshPassthrough := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1 
+kind: MeshPassthrough
+metadata:
+  name: disable-passthrough
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      kuma.io/service: demo-client_mesh-passthrough_svc
+  default:
+    enabled: false
+`, Config.KumaNamespace, meshName)
+
+		// when
+		err := kubernetes.Cluster.Install(YamlK8s(meshPassthrough))
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectFailure(
+				kubernetes.Cluster, "demo-client", "external-service.mesh-passthrough-mes.svc.cluster.local:80",
+				client.FromKubernetesPod(namespace, "demo-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Exitcode).To(Equal(56))
```

Comment:
nitpick: we could have a constant here like `curlRecvError:=56`

```suggestion
			g.Expect(resp.Exitcode).To(Equal(curlRecvError))
```